catalog/solr/catalog-solr-external-provider/src/main/java/ddf/catalog/solr/external/SolrHttpCatalogProvider.java

```diff
@@ -253,14 +259,21 @@ public void updateServer(String urlValue) {
         }
     }
 
+    private void updateServer() {
+        serverFuture = SolrServerFactory.getHttpSolrServer(url,
+                SOLR_CATALOG_CORE_NAME,
+                SOLR_CATALOG_CONFIG_FILE);
+        server = null;
+    }
+
     private CatalogProvider getProvider() {
         if (firstUse) {
```

Comment:
The `firstUse` and `UnconfiguredCatalogProvider` pattern needs to be removed/fixed. As-is, this will get stuck in a failed state if the server is not reachable initially.